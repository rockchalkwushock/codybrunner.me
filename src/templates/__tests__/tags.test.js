import React from 'react'
import { shallow } from 'enzyme'

import TagsTemplate from '../tags'

const pathContext = {
  tags: [['one', [1, 2, 3]], ['two', [1, 2, 3]], ['three', [1, 2, 3]]]
}

describe('Component: <TagsTemplate />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<TagsTemplate pathContext={pathContext} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should receive pathContext.tags as prop', () => {
    expect(wrapper.props()).toHaveProperty('tags')
  })
  test('should receive "tags" as {Array<Array>}', () => {
    expect(wrapper.prop('tags')).toHaveLength(3)
    expect(wrapper.prop('tags')[0]).toHaveLength(2)
    expect(wrapper.prop('tags')[0][0]).toEqual('one')
    expect(wrapper.prop('tags')[0][1]).toHaveLength(3)
  })
})
