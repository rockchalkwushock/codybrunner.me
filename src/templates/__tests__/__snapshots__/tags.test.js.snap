// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TagsTemplate /> should render correctly 1`] = `
<Tags
  tags={
    Array [
      Array [
        "one",
        Array [
          1,
          2,
          3,
        ],
      ],
      Array [
        "two",
        Array [
          1,
          2,
          3,
        ],
      ],
      Array [
        "three",
        Array [
          1,
          2,
          3,
        ],
      ],
    ]
  }
/>
`;
